src/init.cpp

```diff
@@ -1694,7 +1696,7 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     int chain_active_height = WITH_LOCK(cs_main, return chainman.ActiveChain().Height());
 
     // On first startup, warn on low block storage space
-    if (!chainman.m_blockman.m_reindexing && !fReindexChainState && chain_active_height <= 1) {
+    if (!chainman.m_blockman.m_blockfiles_indexed && !do_reindex_chainstate && chain_active_height <= 1) {
```

Comment:
I actually don't understand why we also need to check for `!do_reindex_chainstate` here, when `CChain::vChain` is populated by the blockindex building process, so my intuition tells me `chain_active_height` should only be affected by `-reindex` (but lldb tells me otherwise, so it does indeed seem necessary).

Reply:
> (but lldb tells me otherwise, so it does indeed seem necessary).

This happens because the initial chain tip is set based on the best block in the chainstate database, so if the database is being wiped, the tip is not set and the chain height is 0:

https://github.com/bitcoin/bitcoin/blob/6e4d18f37f42894fe9a0d7948c1da3f061fc6b60/src/node/chainstate.cpp#L136-L138